### DISCUSSION TITLE
How can a parameter name support multiple variable types？

### DISCUSSION THREAD

**Discussion by xxddccaa** [fastapi/fastapi#8800] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# -*- coding:utf-8 -*-
import traceback
from typing import Optional
import uvicorn
from fastapi import FastAPI, File, Form, UploadFile

app = FastAPI()


@app.post("/filesper/")
def filesper(
        file: Optional[UploadFile, bytes, str] = Optional[File(...), Form(...)],
):
    try:
        if file:
            return 1
    except:
        traceback.print_exc()
        return -1


if __name__ == '__main__':
    uvicorn.run('testfastapi:app', host='0.0.0.0', port=8001, reload=False, debug=False, workers=1)
```

### Description

hello!
How can a parameter variable name support multiple types？
In the program，file: Optional[UploadFile, bytes, str] = Optional[File(...), Form(...)]，this program is my imagination，I want file  parameter can be anything that I need, isinstance() can be used to distinguish detail types.
I don't think it can succeed, but I want to know why.
thank you!

### Operating System

Linux

### Operating System Details

CentOS 7

### FastAPI Version

0.68.0

### Python Version

3.

### Additional Context

![](https://raw.githubusercontent.com/xddun/picgo_picture/main/20210823195752.png)

**Comment by xxddccaa** [comment]
Python 3.8.11

**Comment by panla** [comment]
```python
from typing import Union

variable = Union[str, int, dict, float]
variable = Optional[Union[str, int, dict, float]]

```

**Comment by xxddccaa** [comment]
> ```python
> from typing import Union
> 
> variable = Union[str, int, dict, float]
> variable = Optional[Union[str, int, dict, float]]
> ```

thank you! 
Python general types can indeed be written like your code, but url FORM information and FILE information do not seem to be able to do so. I want to express FORM and FILE .

This is a practical application scenario. When I want to pass in a picture or a URL to the url interface, it seems to the user that it is the same thing, so I want to express it with the same parameter.

**Comment by YuriiMotov** [comment, marked as answer]
> file: Optional[UploadFile, bytes, str] = Optional[File(...), Form(...)],

You can not make the same parameter either `Form` or `File` at the same time.
Use different names
